### PR TITLE
GSNSignatureBouncer fix

### PR DESCRIPTION
**Pull request by crazyrabbitLTC**
Ref: https://github.com/OpenZeppelin/openzeppelin-contracts-ethereum-package/pull/67

Current version of the code means the GSNSignatureBouncer will accept Zero Addresses as a trusted signer. The result is that unexpectedly the contract will accept the absence of a signature as a valid signature. The added check prevents the constructor from being called with a zero address.

Fixes #

A require statement was added on the constructor to prevent Zero Address from being supplied as the Trusted Signer. This will prevent the contract being deployed with a Zero Address. 

I added a test to check this functionality.

### CODE REVIEW

File: contracts/GSN/bouncers/GSNBouncerSignature.sol [hunk 1]
```diff
@@ -22,6 +22,7 @@ contract GSNBouncerSignature is GSNBouncerBase {
      * @dev Sets the trusted signer that is going to be producing signatures to approve relayed calls.
      */
     constructor(address trustedSigner) public {
+        require(trustedSigner != address(0), "Trusted Signer can not be a Zero Address");
```
Comment (nventuro) [inline contracts/GSN/bouncers/GSNBouncerSignature.sol]:
```suggestion
        require(trustedSigner != address(0), "GSNBouncerSignature: trusted signer is the zero address");
```

I've updated the revert message to better reflect the general style of the library.

File: test/GSN/GSNBouncerSignature.test.js [hunk 2]
```diff
@@ -17,58 +19,107 @@ contract('GSNBouncerSignature', function ([_, signer, other]) {
     });
   });
 
+  context('when constructor is called with a zero address', function () {
+    it('fails when constructor called with a zero address', async function () {
+      await expectRevert(
+        GSNBouncerSignatureMock.new(
+          '0x0000000000000000000000000000000000000000'
+        ),
+        'Trusted Signer can not be a Zero Address'
```
Comment (nventuro) [inline test/GSN/GSNBouncerSignature.test.js]:
```suggestion
        'GSNBouncerSignature: trusted signer is the zero address'
```

File: test/GSN/GSNBouncerSignature.test.js [hunk 3]
```diff
@@ -17,58 +19,107 @@ contract('GSNBouncerSignature', function ([_, signer, other]) {
     });
   });
 
+  context('when constructor is called with a zero address', function () {
+    it('fails when constructor called with a zero address', async function () {
+      await expectRevert(
+        GSNBouncerSignatureMock.new(
+          '0x0000000000000000000000000000000000000000'
```
Comment (nventuro) [inline test/GSN/GSNBouncerSignature.test.js]: You could use `ZERO_ADDRESS` here:

```
const { constants } = require('openzeppelin-test-helpers');
const { ZERO_ADDRESS } = constants;
```

File: test/GSN/GSNBouncerSignature.test.js [hunk 4]
```diff
@@ -17,58 +19,107 @@ contract('GSNBouncerSignature', function ([_, signer, other]) {
...
+      );
     });
 
     it('rejects relay requests where all parameters are signed by an invalid signer', async function () {
-      const approveFunction = async (data) =>
+      const approveFunction = async data =>
         fixSignature(
           await web3.eth.sign(
             web3.utils.soliditySha3(
               // eslint-disable-next-line max-len
```
Comment (nventuro) [inline test/GSN/GSNBouncerSignature.test.js]: We should remove these if they are no longer needed.

File: contracts/GSN/bouncers/GSNBouncerSignature.sol (hunk 1 above)
Comment (frangio) [reply contracts/GSN/bouncers/GSNBouncerSignature.sol]: Have we written down this convention anywhere?

Comment (nventuro) [reply contracts/GSN/bouncers/GSNBouncerSignature.sol]: There are guidelines in the forum, but I don't think we've added anything to the repo, no.

File: test/GSN/GSNBouncerSignature.test.js (hunk 3 above)
Comment (crazyrabbitLTC) [reply test/GSN/GSNBouncerSignature.test.js]: I did not know about that! Thats really useful!

File: test/GSN/GSNBouncerSignature.test.js (hunk 4 above)
Comment (crazyrabbitLTC) [reply test/GSN/GSNBouncerSignature.test.js]: Ah good point.

File: test/GSN/GSNBouncerSignature.test.js (hunk 3 above)
Comment (crazyrabbitLTC) [reply test/GSN/GSNBouncerSignature.test.js]: Added it.

**Comment by frangio** [comment]
@crazyrabbitLTC I reverted some reformatting that I assume was automatically done by Prettier. Please be careful not to include that sort of thing in the PR, at least unless intended. :slightly_smiling_face:

**Pull request merged**